### DISCUSSION TITLE
Importing Pi4J example gradle app into Eclipse gets Unsupported class file major version 61

### DISCUSSION THREAD

**Discussion by MarkEHansen** [Pi4J/pi4j#278] [Q&A] [answered]
I'm running a Raspberry Pi 4 with the 64-bit Raspberry OS (Raspbian). I've downloaded the AArch64 edition of Eclipse and it runs fine. I'm able to create and run Java applications. I installed Pi4J V2 using "curl -sSL https://pi4j.com/v2/install | sudo bash", which completed successfully. The Pi4J Jar files are found in /opt/pi4j/lib.

I then cloned git@github.com:Pi4J/pi4j-example-minimal.git and was able to build and run using the provided Gradle wrapper and run.sh scripts.

However, when I import the gradle project into Eclipse, I get an error from Gradle "Unsupported class file major version 61".
Version 61 is Java 17. I know this minimal example is wanting Java version 11 (the gradle wrapper is version 6.6, which expects Java version 11).

I've installed OpenJDK 11 using "sudo apt install openjdk-11-jdk" and it is in /var/lib/jvm/java-11-openjdk-arm64.

Although Eclipse comes with OpenJdk 17, I've configured it to use the OpenJdk 11 which I've installed on the machine.

However, when I refresh the Gradle project, I still get the error and the pi4j imports are not found.

**Comment by MarkEHansen** [comment, marked as answer]
Well, it looks like I found the answer. I needed to also set the Java Home in the Advanced Options on the Gradle properties panel in Eclipse. Now it's working as expected.

**Comment by FDelporte** [comment]
Thanks for answering your own question :-)
As I don't have much recent experience myself with Gradle and Eclipse, would you @MarkEHansen be willing to write down how you have set this up with some screenshots etc. to help others? It could be very "drafty" and I can add it to our Pi4J website.

**Reply by MarkEHansen** [reply]
As I mentioned, I'm using the 64-bit Raspberry OS (formerly Raspian). Using this OS, I'm able to install Eclipse for Java Developers, AArch64 edition. Then, to get Eclipse to use the OpenJDK 11, I went to Preferences > Java > Installed JREs and added a new JRE, selecting the "var/lib/jvm/java-11-openjdk-arm64" directory.

I then configured Gradle in Eclipse by going to Preferences > Gradle, and in the Advanced Options section, set the Java Home setting to the same value (/usr/lib/jvm/java-aa-opendjk-arm64).

Now I can import (as a Gradle project) the sample application and Eclipse is able to find the Jars/classes.
